### PR TITLE
Migrate to null-safety

### PR DESCRIPTION
**Pull request by definev**
Migrate dart package to null-safety

### CODE REVIEW

File: transformers/dart-dvhcvn/example/lib/main.dart [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import 'package:dvhcvn/dvhcvn.dart' as dvhcvn;
 
 void main() {
-  final haNoi = dvhcvn.findLevel1ById('01');
-  final baDinh = haNoi.findLevel2ById('001');
+  final haNoi = dvhcvn.findLevel1ById('01')!;
+  final baDinh = haNoi.findLevel2ById('001')!;
   final phucXa = baDinh.findLevel3ById('00001');
```
Comment (daohoangson) [inline transformers/dart-dvhcvn/example/lib/main.dart:6]: I would avoid using `!` which may cause runtime error.

```suggestion
  final haNoi = dvhcvn.findLevel1ById('01');
  final baDinh = haNoi?.findLevel2ById('001');
  final phucXa = baDinh?.findLevel3ById('00001');
```

File: transformers/dart-dvhcvn/lib/src/model.dart [hunk 2]
```diff
@@ -48,7 +48,7 @@ abstract class Entity<ParentType, ChildType> {
 
   @override
   String toString() {
-    final parent = this.parent?.toString();
+    final String? parent = this.parent?.toString();
```
Comment (daohoangson) [inline transformers/dart-dvhcvn/lib/src/model.dart:51]: Let's omit the type here for cleaner code. This is just a personal preference.

```suggestion
    final parent = this.parent?.toString();
```

File: transformers/dart-dvhcvn/lib/src/model.dart [hunk 3]
```diff
@@ -62,10 +62,10 @@ class Level1 extends Entity<void, Level2> {
   void get parent => null;
 
   /// Finds sub-entity by ID.
-  Level2 findLevel2ById(String id) => findById<Level2>(children, id);
+  Level2? findLevel2ById(String id) => findById<Level2>(children, id);
```
Comment (daohoangson) [inline transformers/dart-dvhcvn/lib/src/model.dart:65]: I think we can drop the type specifier here, Dart 2.12 is smart enough to figure this out. Something like this:

```
Level2? findLevel2ById(String id) => findById(children, id);
```

File: transformers/dart-dvhcvn/test/data_test.dart [hunk 4]
```diff
@@ -11,69 +11,69 @@ void main() {
   test('findLevel1ById returns', () {
     final level1 = findLevel1ById('01');
     expect(level1, isNotNull);
-    expect(level1.name, equals('Thành phố Hà Nội'));
+    expect(level1!.name, equals('Thành phố Hà Nội'));
```
Comment (daohoangson) [inline transformers/dart-dvhcvn/test/data_test.dart:14]: Is this really needed?

**Review by daohoangson — CHANGES_REQUESTED** [review]

**Comment by daohoangson** [comment]
Thank you for the PR. I have requested some changes to make it cleaner 😝

File: transformers/dart-dvhcvn/example/lib/main.dart (hunk 1 above)
Comment (definev) [reply transformers/dart-dvhcvn/example/lib/main.dart:6]: Ok it's up to you

**Pull request merged**